src/annotator/anchoring/test/range-test.js

```diff
@@ -0,0 +1,532 @@
+        });
+
+        it('cuts the text node if there is a sibling but text node length > `startOffset`', () => {
+          const browserRange = createBrowserRange({
+            startContainer: container.querySelector('#p-2').firstChild,
+            startOffset: 1,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.equal(normalizedRange.start.data, 'ext 2a');
+        });
+
+        it('does not cut the text node', () => {
+          const browserRange = createBrowserRange({
+            startContainer: container.querySelector('#p-2').firstChild,
+            startOffset: 7,
+            endOffset: 14,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.equal(normalizedRange.start.data, 'text 2b');
+        });
+      });
+
+      context('the whole selection is inside one text element', () => {
+        [
+          {
+            startOffset: 1,
+            endOffset: 3,
+            result: 'ex',
+          },
+          {
+            startOffset: 0,
+            endOffset: 7,
+            result: 'text 1',
+          },
+        ].forEach(test => {
+          it('crops the text node appropriately', () => {
+            const browserRange = createBrowserRange({
+              startContainer: container.querySelector('#p-1').firstChild,
+              endContainer: container.querySelector('#p-1').firstChild,
+              startOffset: test.startOffset,
+              endOffset: test.endOffset,
+            });
+            const normalizedRange = browserRange.normalize();
+            assert.equal(normalizedRange.start.data, test.result);
+          });
+        });
+      });
+
+      context('common ancestor is not ELEMENT_NODE', () => {
+        it('corrects the commons ancestor to the first non text node', () => {
+          const browserRange = createBrowserRange({
+            commonAncestorContainer: container.querySelector('#p-1').firstChild,
+          });
+          const normalizedRange = browserRange.normalize();
+          assert.deepEqual(
+            normalizedRange.commonAncestor,
+            container.querySelector('#p-1')
+          );
+        });
+      });
+    });
+
+    describe('#serialize', () => {
+      it('converts BrowserRange to SerializedRange instance', () => {
+        const browserRange = createBrowserRange();
+        const result = browserRange.serialize(container);
+        assert.isTrue(result instanceof Range.SerializedRange);
+        assert.deepEqual(result, {
+          start: '/section[1]/p[1]',
+          startOffset: 0,
+          end: '/section[1]/span[1]/p[1]',
+          endOffset: 1,
+        });
+      });
+
+      it('converts BrowserRange to SerializedRange instance with `ignoreSelector` condition', () => {
+        const browserRange = createBrowserRange();
+        const result = browserRange.serialize(container, '#p-3');
+        // End xpath shall ignore the provided selector.
```

Comment:
This seems odd. Why is only the end xpath affected?

Reply:
Only because the xpath with #p-3 only relates to the end because that is how its set up `createBrowserRange`, but if we had a common node like <span class="foo"> in the DOM and that was found in both start and end xpaths, then it should ignore that in both the start and end xpaths if ".foo" is passed as the ignoreSelector param.

What I will do is change the ignore selector to just `p` and it will affect both the start and end and make it clear.